fibonacci/cpp/code.cpp

```diff
@@ -1,24 +1,46 @@
-#include <iostream>
-using namespace std;
+//
+// "Modern", more idiomatic C++ version of the naive Fibonacci
+// sequence generator
+//
+// Changes from the C-based version:
+// - Added command line argument count check
+// - C++ does not require "return 0" in the main function
+// - Trailing return types
+// - Convert argv/argc to std::span to void pointer arithmetic
+//   and allow for simpler iteration (if needed)
+// - C++23 provides std::print() and std::println() to replace
+//   std::cout and friends
+// - C++23 ranges and algorithms to clarify intent and to avoid
+//   raw for loops.
+//   See Sean Parents "No raw for loops" and other sources.
+//
+// Minor performance tweak:
+// - Replaced separate if() checks for n == 0 and n == 1 with
+//   single n < 2 check.
+//
 
-int fibonacci(int n)
-{
-    if (n == 0)
-        return 0;
-    if (n == 1)
-        return 1;
+#include <algorithm>
+#include <print>
+#include <ranges>
+#include <span>
+
+[[nodiscard]] constexpr auto fibonacci(int n) -> int {
```

Comment:
Why do you constexpr here?
What do you want to calculate at compile-time?
Do other langs use any compile-time mechanisms? Is it even allowed?

Reply:
Explanation below, but removed it just to remove any doubt.